File: Utils/CRUD_in_list/src/Nodo.cpp
/*
Universidad de las Fuerzas Armadas ESPE
Programa generico para el uso de listas
Alexis Viteri
Fecha de creacion:  24/10/2024
Ultima fecha de modificacion:   24/11/2024
Estructura de Datos
NRC:  1978
*/
#include "../include/Nodo.h"

Nodo::Nodo(int dato) {
    this->dato = dato;
    this->siguiente = nullptr;
}

int Nodo::getDato() {
    return dato;
}

void Nodo::setDato(int dato) {
    this->dato = dato;
}

Nodo* Nodo::getSiguiente() {
    return siguiente;
}

void Nodo::setSiguiente(Nodo* siguiente) {
    this->siguiente = siguiente;
}

File: Assignments/2nd unit/Proyecto_en_python/src/nodo.py
class Nodo:
    def __init__(self, dato):
        self.dato = dato
        self.siguiente = None
        self.anterior = None

    def get_dato(self):
        return self.dato

    def set_dato(self, dato):
        self.dato = dato

    def set_siguiente(self, siguiente):
        self.siguiente = siguiente

    def get_siguiente(self):
        return self.siguiente

    def set_anterior(self, anterior):
        self.anterior = anterior

    def get_anterior(self):
        return self.anterior